plugins/modules/site_workflow_manager.py

```diff
@@ -698,19 +893,34 @@ def site_requires_update(self):
             specified parameters, such as the site type and site details.
         """
 
-        type = self.have['current_site']['type']
```

Comment:
As they are common in both versions, we can keep them outside. Shall we say site_type instead of tyep

    current_site = self.have.get('current_site', {})
    site_type = current_site.get('type')

    self.log("Checking if site requires updates for site type: {}".format(site_type), "INFO")

Reply:
Added these lines in if else conditions